File: Flood Fill/FloodFill.java
class Solution {
    public void dfsFlood(int[][] image, int sr, int sc, int newColor, int val){

        if(sr<0 || sr>image.length-1 || sc<0 || sc>(image[0].length-1)) return;
        
        image[sr][sc] = newColor;
        
        if(sr>0 && image[sr-1][sc] == val) {
            image[sr-1][sc] = newColor; 
            dfsFlood(image, sr-1,sc,newColor, val);           
        }
        if(sr<image.length-1 && image[sr+1][sc] == val){
            image[sr+1][sc] = newColor;
            dfsFlood(image, sr+1, sc, newColor, val);           
        }
        if(sc>0 && image[sr][sc-1] == val){
            image[sr][sc-1] = newColor;
            dfsFlood(image, sr, sc-1, newColor, val);
        } 
        if(sc<image[0].length-1 && image[sr][sc+1] == val){          
            image[sr][sc+1] = newColor;  
            dfsFlood(image, sr, sc+1, newColor, val);            
        }                          
    }
    public int[][] floodFill(int[][] image, int sr, int sc, int newColor) {
        
        if(newColor == image[sr][sc]) return image;
        dfsFlood(image, sr, sc, newColor, image[sr][sc]);
        return image;
        
    }
}

File: flood-fill/flood-fill.cs
public class Solution {
    int color;
    public int[][] FloodFill(int[][] image, int sr, int sc, int newColor) {
        if(image==null||image.Length==0 || image[sr][sc]==newColor)
        {
            return image;
        }
        
        color = image[sr][sc];
        dfs(image,sr,sc,newColor);
        return image;
        
        
    }
​
​
private void dfs(int[][] image,int row,int col,int newcolor)
{
    if(row <0 || col <0 || row >=image.Length || col >=image[0].Length
      || image[row][col]!=color || image[row][col]==newcolor)
    {
        return;
    }
    
    image[row][col] = newcolor;
    List<(int,int)> dirs = new List<(int,int)>(){(-1,0),(1,0),(0,-1),(0,1)}; 
    foreach(var dir in dirs)
    {
        Console.WriteLine(dir);
        int r = row + dir.Item1;;
        int c = col + dir.Item2;
        dfs(image,r,c,newcolor);
    }
}
}
